### PR TITLE
 LIMS-8212: Order: Table configuration: Make sure that you can configure any field in the child table

### PR DESCRIPTION
**Pull request by NouranYoussefMahasseb**
```
c:\users\lenovo\appdata\local\programs\python\python38-32\lib\site-packages\nose\plugins\manager.py:394: RuntimeWarning: Unable to load plugin noseprogressive = noseprogressive:ProgressivePlugin: No module named '
_curses'
  warn("Unable to load plugin %s: %s" % (ep, e),
2020-09-29 18:05:28.133 | INFO     | api_testing.apis.base_api:_get_authorized_session:46 - Get authorized api session.
2020-09-29 18:05:28.136 | INFO     | api_testing.apis.base_api:_get_authorized_session:47 - admin:admin
2020-09-29 18:05:28.857 | INFO     | api_testing.apis.base_api:_get_authorized_session:58 - session ID : 40k_FF0cbk53VYBya1vCnUmjW7hbc85AegrOIaNd8wo .....

DevTools listening on ws://127.0.0.1:50017/devtools/browser/071baa20-106e-46aa-b928-982c49138339
Order: Table configuration: Make sure that you can configure any field in the child table ...
2020-09-29 18:06:03.328 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test105_configure_any_field_in_child_table
2020-09-29 18:06:06.437 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:609 - wait until page is loaded
2020-09-29 18:06:06.749 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-29 18:06:07.414 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-29 18:06:08.128 | INFO     | api_testing.apis.test_unit_api:set_name_configuration_name_only:437 - set test unit configuration
2020-09-29 18:06:08.813 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-29 18:06:08.822 | INFO     | api_testing.apis.orders_api:set_configuration:498 - set order configuration
2020-09-29 18:06:09.540 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-29 18:06:09.545 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test105_configure_any_field_in_child_table:3669 - Unchecking material type and article name checkboxes in configure table
2020-09-29 18:06:09.547 | INFO     | ui_testing.pages.base_pages:open_configure_table:359 - open configure table
2020-09-29 18:06:09.699 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-29 18:06:12.379 | INFO     | ui_testing.pages.base_pages:open_configure_table:359 - open configure table
2020-09-29 18:06:12.491 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-29 18:06:14.457 | DEBUG    | ui_testing.pages.base_pages:sleep_medium:47 - wait up to 2 sec
2020-09-29 18:06:18.255 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test105_configure_any_field_in_child_table:3674 - Checking that unchecked fields disappear from child table
2020-09-29 18:06:19.529 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:609 - wait until page is loaded
2020-09-29 18:06:20.042 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-29 18:06:20.755 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-29 18:06:21.477 | INFO     | ui_testing.pages.base_pages:open_configure_table:359 - open configure table
2020-09-29 18:06:21.637 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-29 18:06:22.974 | DEBUG    | ui_testing.pages.base_pages:sleep_medium:47 - wait up to 2 sec
2020-09-29 18:06:26.948 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test105_configure_any_field_in_child_table:3681 - Checking after rechecking article name and material type reappear in child tab
le
2020-09-29 18:06:26.950 | INFO     | ui_testing.testcases.base_test:tearDown:33 - go to dashboard page
2020-09-29 18:06:28.572 | INFO     | ui_testing.testcases.base_test:tearDown:35 - TearDown.     
ok

----------------------------------------------------------------------
Ran 1 test in 61.539s

OK

```

### CODE REVIEW

File: ui_testing/elements.py [hunk 1]
```diff
@@ -553,6 +553,11 @@
                                'order': 4},
         'edit order header': {'method': 'xpath',
                               'value': '//span[contains(text(),"Edit Order")]'},
+        'material_type_checkbox':{ 'method':'xpath',
+                          'value':'//*[@id="materialType"]/a/span/div/span/label/span'},
+        'article_name_checkbox': {'method':'xpath',
```
Comment (ghada-37) [inline ui_testing/elements.py:558]: use relative xpath :` //li[@id='materialType']//span[@class='checkbox']//span`

File: ui_testing/elements.py [hunk 2]
```diff
@@ -553,6 +553,11 @@
...
+                                  'value':'//*[@id="articleName"]/a/span/div/span/label/span'},
```
Comment (ghada-37) [inline ui_testing/elements.py:559]: use relative xpath : `//li[@id='articleName']//span[@class='checkbox']//span`

File: ui_testing/pages/base_pages.py [hunk 3]
```diff
@@ -645,3 +645,12 @@ def close_connection_with_database(self, db):
     def get_current_year(self):
         current_year = datetime.datetime.now()
         return str(current_year.year)
+
+    def Clicking_on_checkboxes_in_configure_table(self,field_one,field_two,child=False):
```
Comment (ghada-37) [inline ui_testing/pages/base_pages.py:649]: rename it to ` deselect_specific_active_table_configuration` and make replace fields field one and two by : fields = []

File: ui_testing/pages/base_pages.py [hunk 4]
```diff
@@ -645,3 +645,12 @@ def close_connection_with_database(self, db):
...
+        self.open_configure_table()
+        if child:
+            self.base_selenium.click(element='general:configure_child_table')
+        self.base_selenium.click(element=field_one)
```
Comment (ghada-37) [inline ui_testing/pages/base_pages.py:653]: replace it with for loop :
```
for field in feilds:
       check first that feild is selected?
       if selected :
                   self.base_selenium.click(element=field_two)


```
you can use` checkbox_value.is_selected()` to make sure checkbox is selected or not
and                 checkbox_value = "//li[@id="materialType"']//input[@type='checkbox']"

File: ui_testing/pages/orders_page.py [hunk 5]
```diff
@@ -367,3 +367,11 @@ def get_right_menu_options(self):
     def navigate_to_order_active_table(self):
         self.base_selenium.click(element='orders:order_tab')
         self.sleep_medium()
+
+    def get_child_table_headings(self):
+        self.open_child_table(self.result_table()[0])
+        headers = self.base_selenium.get_table_head_elements(element='general:table_child')
+        Child_table_headings = [i.text for i in headers]
```
Comment (ghada-37) [inline ui_testing/pages/orders_page.py:374]: use small letters

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 6]
```diff
@@ -3657,4 +3658,26 @@ def test104_edit_order_page_then_overview(self, edit_case):
         else:
             self.assertFalse(self.order_page.confirm_popup(check_only=True))
             self.info('asserting redirection to active table')
-            self.assertEqual(self.order_page.orders_url, self.base_selenium.get_url())
\ No newline at end of file
+            self.assertEqual(self.order_page.orders_url, self.base_selenium.get_url())
+    def test105_configure_any_field_in_child_table(self):
+
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:3663]: remove unneeded space

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 7]
```diff
@@ -3657,4 +3658,26 @@ def test104_edit_order_page_then_overview(self, edit_case):
...
+        self.info('Unchecking material type and article name checkboxes in configure table')
+        self.base_pages.Clicking_on_checkboxes_in_configure_table(field_one='orders:material_type_checkbox',field_two='orders:article_name_checkbox',child=True)
+
+        Fields_displayed_in_child_table=self.orders_page.get_child_table_headings()
+        self.info('Checking that unchecked fields disappear from child table')
+        self.assertNotIn('Article Name',Fields_displayed_in_child_table)
+        self.assertNotIn('Material Type',Fields_displayed_in_child_table)
+
+        self.orders_page.get_orders_page()
+        self.base_pages.Clicking_on_checkboxes_in_configure_table(field_one='orders:material_type_checkbox',field_two='orders:article_name_checkbox',child=True)
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:3679]: here you need to deselct changed by not article or material

**Review by ghada-37 — CHANGES_REQUESTED** [review]

File: .github/workflows/execution.sh [hunk 8]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 EXECUTION_FILES=(
-     ui_testing/testcases/extended_tests/test001_order.py
+     ui_testing/testcases/basic_tests/test006_orders.py
   )
 
  TEST_REG='test004'
```
Comment (ghada-37) [inline .github/workflows/execution.sh:7]: change it to 105 to run your test case

File: ui_testing/elements.py (hunk 1 above)
Comment (NouranYoussefMahasseb) [reply ui_testing/elements.py:558]: done

File: ui_testing/elements.py (hunk 2 above)
Comment (NouranYoussefMahasseb) [reply ui_testing/elements.py]: done

File: ui_testing/pages/base_pages.py (hunk 4 above)
Comment (NouranYoussefMahasseb) [reply ui_testing/pages/base_pages.py]: done but the  is selected didn't work always returns False

File: ui_testing/pages/orders_page.py (hunk 5 above)
Comment (NouranYoussefMahasseb) [reply ui_testing/pages/orders_page.py]: done

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 6 above)
Comment (NouranYoussefMahasseb) [reply ui_testing/testcases/basic_tests/test006_orders.py]: done

**Pull request closed without merging**